# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1511296532 -3600
#      Tue Nov 21 21:35:32 2017 +0100
# Node ID ff890a84b43c44a786e3f57145a369d8c7c37eb4
# Parent  ad0ca13aa6e2509ad13fceff6c65bcdcec5cc567
allow uncommenting in editor when line begins with whitespace (bug #52406)

* file-editor-tab.cc (do_comment_selected_text): use a regular expression
  instead of only checking the beginning of the line for a comment string

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -1458,16 +1458,19 @@ file_editor_tab::do_smart_indent_line_or
 
   _edit_area->endUndoAction ();
 }
 
 void
 file_editor_tab::do_comment_selected_text (bool comment)
 {
   QString comment_str = _edit_area->comment_string ();
+  QRegExp rxc = QRegExp ("^([ \\t]*)" + comment_str);
+  int len, lenc = comment_str.length ();
+
   _edit_area->beginUndoAction ();
 
   if (_edit_area->hasSelectedText ())
     {
       int lineFrom, lineTo, colFrom, colTo;
       _edit_area->getSelection (&lineFrom, &colFrom, &lineTo, &colTo);
 
       if (colTo == 0)  // the beginning of last line is not selected
@@ -1475,19 +1478,20 @@ file_editor_tab::do_comment_selected_tex
 
       for (int i = lineFrom; i <= lineTo; i++)
         {
           if (comment)
             _edit_area->insertAt (comment_str, i, 0);
           else
             {
               QString line (_edit_area->text (i));
-              if (line.startsWith (comment_str))
+              if (line.contains (rxc))
                 {
-                  _edit_area->setSelection (i, 0, i, comment_str.length ());
+                  len = rxc.matchedLength ();
+                  _edit_area->setSelection (i, len-lenc, i, len);
                   _edit_area->removeSelectedText ();
                 }
             }
         }
       //set selection on (un)commented section
       _edit_area->setSelection (lineFrom, 0, lineTo,
                                 _edit_area->text (lineTo).length ()-1);
     }
@@ -1495,19 +1499,20 @@ file_editor_tab::do_comment_selected_tex
     {
       int cpline, col;
       _edit_area->getCursorPosition (&cpline, &col);
       if (comment)
         _edit_area->insertAt (comment_str, cpline, 0);
       else
         {
           QString line (_edit_area->text (cpline));
-          if (line.startsWith (comment_str))
+          if (line.contains (rxc))
             {
-              _edit_area->setSelection (cpline, 0, cpline, comment_str.length ());
+              len = rxc.matchedLength ();
+              _edit_area->setSelection (cpline, len-lenc, cpline, len);
               _edit_area->removeSelectedText ();
             }
         }
     }
   _edit_area->endUndoAction ();
 }
 
 void
